packages/core/src/theme.ts

```diff
@@ -57,3 +58,15 @@ export function createTheme<const T extends ThemeInput>(
     components: components,
   } as unknown as ThemeResult<T>;
 }
+
+type ComponentThemeInput<Variants extends { [key: string]: StyledProps }> = {
+  baseStyle?: StyledProps;
+  variants?: Variants;
+  defaultProps?: StyledProps & { variant?: keyof Variants };
```

Comment:
There are use cases where you'd want to pass HTML attributes like WAI-ARIA, so having a type like `Record<string, any>` might be appropriate?

Reply:
True! Also `StyleProps` was unnecessary since `baseStyle` handle them. [a68aa1f](https://github.com/kuma-ui/kuma-ui/pull/306/commits/a68aa1f036895e4c5cb4465a62c3a1bc730882ff)